ui/src/components/admin/ModelsManager/ModelManagerController.js

```diff
@@ -115,350 +120,356 @@ Model table state transitions for each action:
-                action: 'publish',
-                baseUrl,
-                state,
-                setState,
-                appendNotification,
-              }),
-              bulkUnpublish: bulkActionCreator({
-                action: 'unpublish',
-                baseUrl,
-                state,
-                setState,
-                appendNotification,
-              }),
-              bulkDelete: bulkActionCreator({
-                action: 'delete',
-                baseUrl,
-                state,
-                setState,
-                appendNotification,
-              }),
-              publishOne: async name => {
-                // Set loading true (lock UI)
-                await setState({
-                  isLoading: true,
-                });
-
-                singleAction('publish', name)
-                  .then(() => loadData(baseUrl, state))
-                  .then(async ([dataResponse, countResponse]) => {
+                    await publishBulk(state.selection)
+                      .then(async response => {
+                        if (response.data.success) {
+                          await setPublishProgress({
+                            ...publishProgress,
+                            running: true,
+                          });
+                          await tempNotification.clear();
+                        }
+                      })
+                      .catch(async error => {
+                        await appendNotification({
+                          type: NOTIFICATION_TYPES.ERROR,
+                          message: 'Bulk Publish Failed.',
+                          details: error.response
+                            ? error.response.data.error.message
+                            : error.message,
+                          timeout: false,
+                        });
+                        await tempNotification.clear();
+                      });
+                  },
+                  bulkUnpublish: bulkActionCreator({
+                    action: 'unpublish',
+                    baseUrl,
+                    state,
+                    setState,
+                    appendNotification,
+                  }),
+                  bulkDelete: bulkActionCreator({
+                    action: 'delete',
+                    baseUrl,
+                    state,
+                    setState,
+                    appendNotification,
+                  }),
+                  publishOne: async name => {
+                    await publish(name)
+                      .then(async () => {
+                        await addPublishNotification(name);
+                      })
+                      .catch(async error => {
+                        const data = error.response ? error.response.data : error;
+                        showErrorPublishNotification(name, data);
+                      });
+                  },
+                  unpublishOne: async name => {
```

Comment:
Any other changes to this file from here onward are just indentation/spacing changes as a result of wrapping this in another context consumer.